build/source/engine/run_oneGRU.f90

```diff
@@ -219,29 +224,35 @@ subroutine run_oneGRU(&
    end if  ! (if identified a downslope HRU)
   end do dsHRU
 
-  ! add inflow to the downslope HRU
+  ! if lateral flows are active, add inflow to the downslope HRU
   if(kHRU > 0)then  ! if there is a downslope HRU
    fluxHRU%hru(kHRU)%var(iLookFLUX%mLayerColumnInflow)%dat(:) = fluxHRU%hru(kHRU)%var(iLookFLUX%mLayerColumnInflow)%dat(:)  + fluxHRU%hru(iHRU)%var(iLookFLUX%mLayerColumnOutflow)%dat(:)
 
-  ! increment basin column outflow (m3 s-1)
+  ! otherwise just increment basin (GRU) column outflow (m3 s-1) with the hru fraction
   else
-   bvarData%var(iLookBVAR%basin__ColumnOutflow)%dat(1)   = bvarData%var(iLookBVAR%basin__ColumnOutflow)%dat(1) + sum(fluxHRU%hru(iHRU)%var(iLookFLUX%mLayerColumnOutflow)%dat(:))
+   bvarData%var(iLookBVAR%basin__ColumnOutflow)%dat(1) = bvarData%var(iLookBVAR%basin__ColumnOutflow)%dat(1) + sum(fluxHRU%hru(iHRU)%var(iLookFLUX%mLayerColumnOutflow)%dat(:))
   end if
 
+  ! ----- calculate weighted basin (GRU) fluxes --------------------------------------------------------------------------------------
+  
+  ! increment basin total runoff (m s-1)
+  bvarData%var(iLookBVAR%basin__TotalRunoff)%dat(1)    = bvarData%var(iLookBVAR%basin__TotalRunoff)%dat(1)     + fluxHRU%hru(iHRU)%var(iLookFLUX%scalarTotalRunoff)%dat(1)    * fracHRU
+
   ! increment basin surface runoff (m s-1)
-  bvarData%var(iLookBVAR%basin__SurfaceRunoff)%dat(1)    = bvarData%var(iLookBVAR%basin__SurfaceRunoff)%dat(1)     + fluxHRU%hru(iHRU)%var(iLookFLUX%scalarSurfaceRunoff)%dat(1)    * fracHRU
+  bvarData%var(iLookBVAR%basin__SurfaceRunoff)%dat(1)  = bvarData%var(iLookBVAR%basin__SurfaceRunoff)%dat(1)     + fluxHRU%hru(iHRU)%var(iLookFLUX%scalarSurfaceRunoff)%dat(1)    * fracHRU
 
   ! increment basin-average baseflow input variables (m s-1)
   bvarData%var(iLookBVAR%basin__AquiferRecharge)%dat(1)  = bvarData%var(iLookBVAR%basin__AquiferRecharge)%dat(1)   + fluxHRU%hru(iHRU)%var(iLookFLUX%scalarSoilDrainage)%dat(1)     * fracHRU
   bvarData%var(iLookBVAR%basin__AquiferTranspire)%dat(1) = bvarData%var(iLookBVAR%basin__AquiferTranspire)%dat(1)  + fluxHRU%hru(iHRU)%var(iLookFLUX%scalarAquiferTranspire)%dat(1) * fracHRU
 
-  ! increment aquifer baseflow -- ONLY if baseflow is computed individually for each HRU
+  ! increment aquifer baseflow -- ONLY if aquifer baseflow is computed individually for each HRU and aquifer is run
   ! NOTE: groundwater computed later for singleBasin
-  if(model_decisions(iLookDECISIONS%spatial_gw)%iDecision == localColumn)then
+  if(model_decisions(iLookDECISIONS%spatial_gw)%iDecision == localColumn .and. model_decisions(iLookDECISIONS%groundwatr)%iDecision == bigBucket) then
```

Comment:
Do we need the additional condition that the `bigBucket` must be run? `aquiferBaseflow` is set to zero if there is no aquifer. Keeping the original code ensures that `basin__AquiferBaseflow` is also set to zero.

Reply:
I'm not sure - I was thinking we don't need to calculate it if it's zero (ie not run), and it's initialized to zero higher up (before the hru loop).

perhaps the two other aquifer summation variables above the if condition should also be included in this block.

Reply:
Yes, we could go either way I suppose. It could be easier to maintain if we reduce the number of conditions.

Reply:
I agree it is OK to add the if condition for `bigBucket`